optimization debian package manager tweaks

rajpratik71 opened the pull request:
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=22652) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=22652) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=22652) it.</sub>

oddlittlebird commented:
@xlson , please review.

xlson reviewed (commented):
Thanks for contribution. I'll take a look at this when I get some time over, but it's not very high on the priority list as it will take quite some time to verify that we don't depend on one of the recommended packages.

@Rajpratik71 if you want to make sure this gets reviewed and merged sooner it would be a good idea to break this apart into a few different PRs. One for the main Dockerfiles (`Dockerfile`, `packaging/docker/*`) that we use to build grafana, one for the docs and one for the ci docker containers (`scripts/build/ci-*`).

aknuds1 commented:
Please break this into several PRs, since it touches on too many different files (both Dockerfiles for internally used images and externally used images, and documentation).

Rajpratik71 commented:
@aknuds1 requesting review

Rajpratik71 commented:
for the failing build issue , i looked at logs , it is related to "TypeScript" which is not touched means unrelated failing builds

oddlittlebird commented:
Try updating the branch from master.

Rajpratik71 commented:
> Try updating the branch from master.

Updated to upstream . Please review . Is pending from months

Rajpratik71 commented:
@xlson @oddlittlebird @aknuds1  requesting review

aknuds1 commented on .circleci/config.yml:
```diff
@@ -735,25 +735,25 @@ jobs:
-          name: ci job succeeded
-          command: "./scripts/ci-job-succeeded.sh"
-          when: on_success
+        - checkout
```
Please revert your formatting changes.

Rajpratik71 replied:
fixed

aknuds1 commented on Dockerfile.ubuntu:
```diff
@@ -49,8 +49,8 @@ WORKDIR $GF_PATHS_HOME
 
-# curl should be part of the image
-RUN apt-get update && apt-get upgrade -y && apt-get install -y ca-certificates curl
+# We need font libs for phantomjs, and curl should be part of the image
```
Please revert this change.

Rajpratik71 replied:
fixed

aknuds1 commented on scripts/build/ci-build/Dockerfile:
```diff
@@ -56,8 +56,8 @@ RUN UNATTENDED=yes OSX_VERSION_MIN=${OSX_MIN} /tmp/osxcross/build.sh
     tar cfJ osxcross.tar.xz osxcross/target              && \
-    rm -rf /tmp/osxcross
-RUN apt-get install -y                     \
+    rm -rf /tmp/osxcross && \
```
Please revert this change.

Rajpratik71 replied:
fixed

aknuds1 commented:
Just a cursory review for now. In order to review in depth, I need to find the time, since I have to verify that your changes don't break things.

Rajpratik71 commented:
> for the failing build issue , i looked at logs , it is related to "TypeScript" which is not touched means unrelated failing builds

Again this happened after rebasing from Upstream

Rajpratik71 commented:
@xlson @aknuds1 @oddlittlebird  requesting review

aknuds1 commented on .circleci/config.yml:
```diff
@@ -728,7 +728,7 @@ jobs:
           command: "./scripts/ci-job-started.sh"
       - run: sudo apt update
-      - run: sudo apt install -y default-mysql-client
+      - run: sudo apt-get --no-install-recommends install -y default-mysql-client
```
Please remove this change.

aknuds1 commented on .circleci/config.yml:
```diff
@@ -757,7 +757,7 @@ jobs:
           command: "./scripts/ci-job-started.sh"
       - run: sudo apt update
-      - run: sudo apt install -y postgresql-client
+      - run: sudo apt-get --no-install-recommends install -y postgresql-client
```
Please remove this change.

aknuds1 commented on Dockerfile.ubuntu:
```diff
@@ -50,7 +50,7 @@ WORKDIR $GF_PATHS_HOME
 
 # curl should be part of the image
-RUN apt-get update && apt-get install -y ca-certificates curl
+RUN apt-get update && apt-get upgrade -y && apt-get --no-install-recommends install -y ca-certificates curl
```
```suggestion
RUN apt-get update && apt-get upgrade -y && apt-get install --no-install-recommends -y ca-certificates curl
```

aknuds1 commented on devenv/docker/blocks/multiple-openldap/admins-ldap-server.Dockerfile:
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Please remove this change.

aknuds1 commented on devenv/docker/blocks/multiple-openldap/ldap-server.Dockerfile:
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Please remove this change.

aknuds1 commented on devenv/docker/blocks/mysql_opendata/Dockerfile:
```diff
@@ -8,7 +8,7 @@ ENV MYSQL_DATABASE="testdata" \
 
 # Install requirement (wget)
-RUN apt-get update && apt-get install -y wget && apt-get install unzip
+RUN apt-get update && apt-get --no-install-recommends install -y wget && apt-get --no-install-recommends install -y unzip
```
Please remove this change.

aknuds1 commented on devenv/docker/blocks/openldap/Dockerfile:
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Please remove this change.

aknuds1 commented on devenv/docker/debtest/Dockerfile:
```diff
@@ -1,6 +1,6 @@
 FROM debian:jessie
 
-RUN apt-get update && apt-get install -y vim
+RUN apt-get update && apt-get --no-install-recommends install -y vim
```
Please remove this change.

aknuds1 commented on devenv/docker/ha_test/README.md:
````diff
@@ -26,7 +26,7 @@ $ make build-docker-full
 
 ```bash
-$ sudo apt-get install dnsmasq
+$ sudo apt-get --no-install-recommends install -y dnsmasq
````
Please remove this change.

aknuds1 commented on docs/sources/installation/debian.md:
````diff
@@ -36,8 +36,8 @@ If you install from the APT repository, then Grafana is automatically updated ev
 ```bash
-sudo apt-get install -y apt-transport-https
-sudo apt-get install -y software-properties-common wget
+sudo apt-get --no-install-recommends install -y apt-transport-https
+sudo apt-get --no-install-recommends install -y software-properties-common wget
````
Please remove these changes.

aknuds1 commented on docs/sources/installation/debian.md:
````diff
@@ -56,14 +56,14 @@ After you add the repository:
-sudo apt-get install grafana-enterprise
+sudo apt-get --no-install-recommends install -y grafana-enterprise
 ```
 
 #### To install the latest OSS release:
 
 ```bash
-sudo apt-get install -y apt-transport-https
-sudo apt-get install -y software-properties-common wget
+sudo apt-get --no-install-recommends install -y apt-transport-https
+sudo apt-get --no-install-recommends install -y software-properties-common wget
````
Please remove these changes.

aknuds1 commented on docs/sources/installation/debian.md:
````diff
@@ -82,7 +82,7 @@ After you add the repository:
 ```bash
 sudo apt-get update
-sudo apt-get install grafana
+sudo apt-get --no-install-recommends install grafana
````
Please remove this change.

aknuds1 commented on docs/sources/installation/debian.md:
````diff
@@ -99,7 +99,7 @@ If you install the `.deb` package, then you will need to manually update Grafana
 
 ```bash
-sudo apt-get install -y adduser libfontconfig1
+sudo apt-get --no-install-recommends install -y adduser libfontconfig1
````
Please remove this change.

aknuds1 commented on docs/sources/installation/upgrading.md:
````diff
@@ -65,7 +65,7 @@ links.
 ```bash
 wget <debian package url>
-sudo apt-get install -y adduser libfontconfig1
+sudo apt-get --no-install-recommends install -y adduser libfontconfig1
````
Please remove this change.

aknuds1 commented on docs/sources/tutorials/stack_guide_graphite.md:
````diff
@@ -39,7 +39,7 @@ queries that will make the most out of your collected metrics.
 
 ```
-apt-get install \
+apt-get --no-install-recommends install -y \
````
Please remove this change.

aknuds1 commented on docs/sources/tutorials/stack_guide_graphite.md:
```diff
@@ -146,7 +146,7 @@ end with max, min or count then use average. This means that naming metrics is v
 Lets install supervisord and let it start carbon.
 
-`apt-get install supervisor`
+`apt-get --no-install-recommends install -y supervisor`
```
Please remove this change.

aknuds1 commented on packaging/docker/ubuntu.Dockerfile:
```diff
@@ -28,7 +28,7 @@ WORKDIR $GF_PATHS_HOME
 # Install dependencies
 # We need curl in the image
-RUN apt-get update && apt-get install -y ca-certificates curl tzdata && \
+RUN apt-get update && apt-get --no-install-recommends install -y ca-certificates curl tzdata && \
```
```suggestion
RUN apt-get update && apt-get install --no-install-recommends  -y ca-certificates curl tzdata && \
```

aknuds1 commented on scripts/build/ci-build/Dockerfile:
```diff
@@ -35,7 +35,7 @@ WORKDIR /tmp
 
 RUN apt-get update && \
-    apt-get install -yq \
+    apt-get --no-install-recommends install -yq \
```
Please remove this change.

aknuds1 commented on scripts/build/ci-build/Dockerfile:
```diff
@@ -54,7 +54,8 @@ RUN rm -rf /tmp/osxcross/target/SDK/${OSX_SDK}/usr/share && \
     rm -rf /tmp/osxcross
-RUN apt-get install -yq                     \
+
+RUN apt-get --no-install-recommends install -yq                     \
```
Please remove this change.

aknuds1 commented on scripts/build/ci-build/Dockerfile:
```diff
@@ -99,7 +100,7 @@ COPY --from=toolchain /tmp/x86_64-centos6-linux-gnu.tar.xz /tmp/osxcross.tar.xz
 
 RUN apt-get update && \
-    apt-get install -yq  \
+    apt-get --no-install-recommends install -yq  \
```
Please remove this change.

aknuds1 commented on scripts/build/ci-deploy/Dockerfile:
```diff
@@ -14,7 +14,7 @@ RUN pip install -U awscli crcmod && \
       tar xvzf - -C /opt && \
     apt update && \
-    apt install -y createrepo expect && \
+    apt-get --no-install-recommends install -y createrepo expect && \
```
Please remove this change.

aknuds1 commented on scripts/verify-repo-update/Dockerfile.deb:
```diff
@@ -4,12 +4,12 @@ ARG REPO_CONFIG=grafana.list.oss
 
 RUN apt update && \
-    apt install -y curl                \
+    apt-get --no-install-recommends install -y curl                \
```
Please remove this change.

aknuds1 commented on scripts/verify-repo-update/Dockerfile.deb:
```diff
@@ -4,12 +4,12 @@ ARG REPO_CONFIG=grafana.list.oss
 COPY "./$REPO_CONFIG" /etc/apt/sources.list.d/grafana.list
 RUN apt update && \
-    apt install -y $PACKAGE
+    apt-get --no-install-recommends install -y $PACKAGE
```
Please remove this change.

aknuds1 reviewed (changes requested):
We've decided that most changes from the PR should be dropped.

Rajpratik71 commented:
@aknuds1 removed everyone which is not needed or was duplicate. Only left with where needed.

aknuds1 commented:
Why have you resolved my review comments? Please do as I ask in the comments, or I will have to close the PR.

aknuds1 commented on devenv/docker/blocks/mysql_opendata/Dockerfile:
```diff
@@ -8,7 +8,7 @@ ENV MYSQL_DATABASE="testdata" \
 
 # Install requirement (wget)
-RUN apt-get update && apt-get install -y wget && apt-get install unzip
+RUN apt-get update && apt-get --no-install-recommends install -y wget unzip
```
Please remove this change.

Rajpratik71 replied:
done as requested

rajpratik71 committed:
optimization debian package manager tweaks

By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

aknuds1 reviewed (changes requested):
Please do as I ask in review comments.

Rajpratik71 commented:
removing everything leave nothing in commit , closing it .
